src/frontend/src/utils/countdown.ts

```diff
@@ -100,3 +101,80 @@ const calculateTimeRemaining = (
       .padStart(2, "0"),
   ];
 };
+
+// Prettify the number into a string: "mm:ss"
+function prettifySeconds(a: number): string {
+  const mins = Math.floor(a / 60)
+    .toString()
+    .padStart(2, "0");
+  const secs = Math.floor(a % 60)
+    .toString()
+    .padStart(2, "0");
+  return `${mins}:${secs}`;
+}
+
+/// A one shot countdown timer that emits the remaining time as a formatted string ("mm:ss")
+// through an AsyncIterable.
+//
+// The countdown cannot be restarted.
+export class AsyncCountdown {
+  // Create a countdown from nanoseconds
+  static fromNanos(nanos: bigint) {
+    return new AsyncCountdown(Number(nanos / BigInt(1e9)));
+  }
+
+  // The number of seconds elapsed since the epoch
+  static seconds() {
+    return new Date().getTime() / 1000;
```

Comment:
should it be rounded?

Reply:
Definitely not! we use the fractional part to schedule a delay